iroh-bytes/src/get.rs

```diff
@@ -491,6 +494,16 @@ pub mod fsm {
             Ok(res)
         }
 
+        /// Write the entire stream for this blob to a batch writer.
+        pub async fn write_all_batch<B>(self, batch: B) -> result::Result<AtEndBlob, DecodeError>
+        where
+            B: BaoBatchWriter,
+        {
+            let (content, _size) = self.next().await?;
```

Comment:
would it make sense to report the total written size in return value?

Reply:
This returns a state of the iroh-bytes state machine. If you follow the state machine to the end without an error, you get a Stats object which contains this information.

But it is hard to come up with stats that are not too expensive to compute and that are useful for everybody. So to get very precise info you would have to write a wrapper for the batch writer, just like the FallibleProgressBatchWriter, that computes some detailed stats.

```
pub struct Stats {
    /// The number of bytes written
    pub bytes_written: u64,
    /// The number of bytes read
    pub bytes_read: u64,
    /// The time it took to transfer the data
    pub elapsed: Duration,
}
```